rest/config_legacy.go

```diff
@@ -274,14 +274,14 @@ func gocbv1tov2AddressUpgrade(server string) (newServer, username, password stri
 
 	var hosts string
 	for _, addr := range connSpec.Addresses {
+		if addr.Port != -1 && addr.Port != 8091 && connSpec.Scheme == "http" {
```

Comment:
Unset ports with `http://` are an implicit port `80`, which I think is safe to assume is a non-default Couchbase port, and should error, because we can't make the assumption that the memcached port is default.